<html>
<body>
<head>
    <style>
      html * {
        font-family: arial;
      }
	.help-icon {
	content: url(data:image/gif;base64,iVBORw0KGgoAAAANSUhEUgAAACUAAAAjCAIAAACcpVRJAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAGNSURBVFhHxZU7boNAFEWz5KyDMkugYA0WRXooESWSW4RE5548Zq7HzJ3/SI6vTuO8z4mHIfk6svP98xsCHRlJ+2h1HMyEE/PRrnww70vQRytKwRYnHh9Nappu7pfHuqNHRT5u/Tg3TrMBjZewjwZOuvtkaTxZl6AVHc9YPmoVmtuGWjL71jrjGjSovHzUdJIv09nvyW8Z8c0TKiaP6Ta8Nqonisoz6zhcNrxA2fioLDSjvSvwu3PbkTjVkG/o7avYd9Rg4GOYbtQAdPX0UUFhbwk/GKFd0KUT8glSDfnyoZN4s895YSInr3z0oyKcy3Icy0w9RLVvaJ2XIXI5DXU+fmYqWxs+SUOFzycL/zEjin10+89E3xai0NfdV0gQ+c/APVHKfHwhC2VCmc8+zNxndqXofbdvSslj08jQJ3wSKryJf/VpEXwSKvuoP0+MfNInoSaHSh/6VSyfhFptanxofoZ9Ehq4UOxD5yUen4TG6sAuO36fDs3ng3lfYj4d2hUHM+GkfSa0+go6kjmOP/BCFUG6Xal8AAAAAElFTkSuQmCC) ;
	width: 15;
	height: 15;
	}	  
    <style type="text/css">
    .table_header {
        font-weight: bold;
    }
    .header_line {
        font-weight: bold;
		font-size: 18px;
    }
	table, th, td {
		border: 1px solid black;
	}
	.diagram {
		line-height: 0.2;
		font-family: Courier;
		letter-spacing: 0px;
	}		
	td {
	  max-width:250px;
	  text-align: center;
	  word-wrap: break-word;
	}
	.td_alt {
	  max-width:250px;
	  text-align: left;
	  word-wrap: break-word;
	}	
	.td_alt_wide {
	  max-width:450px;
	  text-align: left;
	  word-wrap: break-word;
	}	
	.alt_row{
		background-color:#b0e0e6;
	}

    </style>
</head>
<div class=header_line><a name="Summary">Summary of results</div>
<div>Model name: typhi_v8, model date: 18/Dec/24</div>
<br/>
<table>
	<tbody>
<tr class=table_header><td>Sample  <img class="help-icon" title="Sample name, you can make it more informative by adding sample description using options -c and -d."></td><td>Amplicons by # of dominant unknown SNPs  <img class="help-icon" title="Numbers means number of amplicons, number's position-1 indicates how many unknown SNPs these amplicons have.&#10E.g. 2,0,4,0,0 means there are 2 amplicons without unknown SNPs (position is 1, so 1-1=0), and 4 amplicons with 2 unknown SNPs each (position is 3-1=2).&#10Only examines the amplicons that should be always present i.e. excludes plasmids, recombinations, etc."></td><td>Possible High Level Genotypes  <img class="help-icon" title="Relevant for some organisms (e.g. S. Typhi) where some genotypes cannot be determined with single SNP.&#10Shows high level genotypes (in S. Typhi these are 0, 1, 2, 3, 4) that are possible given detected amplicons.&#10Uses all alleles that are supported by >50reads.&#10'Any' means detected amplicons provide no information on the high level genotypes."></td><td>Implied Genotypes  <img class="help-icon" title="Shows only the most specific genotype based on detected amplicons. Will not show multiple genotypes even if sample consists of more than one."></td><td>Implied AMR  <img class="help-icon" title="Blank field means AMR amplicons not detected,&#10Amplicon name means gene detected, but doesn't have AMR mutations,&#10Mutation description means this mutation was detected in respective amplicon"></td><td>Sequencing Reads  <img class="help-icon" title="Total reads in FASTQ files. If classification is done from BAM files this field will be blank."></td><td>Mapped, correct length  <img class="help-icon" title="Reads which mapped to some amplicon and have length that meets -l and -s parameters."></td><td>Mapped, but too short/long  <img class="help-icon" title="Reads which mapped to some amplicons, but have length that does not meet -l and -s parameters.&#10See values used just below 'Summary of results' table"></td></tr>
<tr class=alt_row><td><a href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>2,&nbsp;0,&nbsp;0,&nbsp;0,&nbsp;0,&nbsp;0</td><td>-</td><td>-</td><td>-</td><td>32167</td><td>3015</td><td>820</td></tr>
<tr><td><a href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>11,&nbsp;1,&nbsp;0,&nbsp;0,&nbsp;0,&nbsp;0</td><td>4</td><td>4.3.1.1 (94%)</td><td>gyrA_S83T (99%)</td><td>14187</td><td>2534</td><td>983</td></tr>
	</tbody>
</table>
<p>"-" fewer than 2 amplicons with 50 target organism reads and fewer than 2 SNPs</p><p>Classifier allowed soft clip (-s) of 25nt and 10nt length difference (-l) between reference and mapped read.<p>Report generated on 18/Dec/24 at 14:34. <a href="#ModelSignatures">See models list</a></p><br/><br/>
<div class=header_line><a name="barcode1">barcode1<br/>S. Typhi negative wastewater</a><div><a href="#Summary">Back to Summary</a></div></div>
<br/>
<table>
	<tbody>
<tr class=table_header><td>Amplicon  <img class="help-icon" title="Sample name, you can make it more informative by adding sample description using options -c and -d."></td><td>Tot. Reads  <img class="help-icon" title="Total number of reads that mapped to the amplicon regadless of length (i.e. -s and -l parameters)"></td><td>Rejected on Length  <img class="help-icon" title="Number of mapped reads that were not assesed by classifier due to not meeting -s and -l parameters."></td><td>% Target Org.  <img class="help-icon" title="Percentage of mapped reads that were classified as target organism and that satisfied -s and -l length parameteres"></td><td># Target Org.  <img class="help-icon" title="Number of reads that were classified as target organism and that satisfied -s and -l length parameteres"></td><td>All known SNPs  <img class="help-icon" title="Number of SNPs detected that match known SNPs defined in classifier model file you used.&#10Includes all alleles different from references that occur in >50 reads and no less than 0.05% of all reads"></td><td>Dominant known SNPs  <img class="help-icon" title="Shows those detected SNPs that match known SNPs in classifier model file you used and also are the most common nucleotide that position.&#10E.g. if at position 254 the reference is C, but in reads the most common nucleotide is T this would count in this column.&#10If T occcured, but the most common was C, this would not count in this column. "></td><td>Dominant unknown SNPs  <img class="help-icon" title="Shows those detected SNPs that DO NOT match known SNPs in classifier model file you used and also are the most common nucleotide that position.&#10E.g. if at position 315 the reference is A, but in reads the most common nucleotide is G this would count in this column.&#10If G occcured, but the most common was A, this would not count in this column. If position has three occuring nucleotides, only the most common is examined."></td><td>Implied Genotypes  <img class="help-icon" title="Genotypes implied by detected SNPs (but not reference alleles). Number in brackets indicates what % of target organism reads supports this genotype.&#10For high level genotypes, a range of possible gnotypes (e.g. 0-2 or 3-4) is reported."></td><td>Implied AMR  <img class="help-icon" title="AMR genotype implied by detected SNPs (but not reference alleles). Number in brackets indicates what % of target organism reads supports this genotype."></td></tr>

<tr class=alt_row><td><a href="#barcode1_1_3_4.3.1_consensus"/>1_3_4.3.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode1_2.2.2_2_consensus"/>2.2.2_2</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode1_2.3.1_v12_consensus"/>2.3.1_v12</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode1_2.3.2_consensus"/>2.3.2</td><td>2215</td><td>447</td><td>99.66%</td><td>1762</td><td>0</td><td>0</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a href="#barcode1_2.5_4.3.1.2.1_3.3_consensus"/>2.5_4.3.1.2.1_3.3</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode1_3.1.1_v5_consensus"/>3.1.1_v5</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode1_3.3.1_consensus"/>3.3.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode1_4.3.1.1_consensus"/>4.3.1.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode1_4.3.1.1.P1_v2_consensus"/>4.3.1.1.P1_v2</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode1_4.3.1.2.1.1_4_consensus"/>4.3.1.2.1.1_4</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode1_4.3.1.2_v3_consensus"/>4.3.1.2_v3</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode1_parC_v7_consensus"/>parC_v7</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode1_tviD_v2_consensus"/>tviD_v2</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode1_acrB_v2_consensus"/>acrB_v2</td><td>1463</td><td>311</td><td>99.83%</td><td>1150</td><td>0</td><td>0</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a href="#barcode1_gyrA_consensus"/>gyrA</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode1_plasmid_ST6_C19241A_consensus"/>plasmid_ST6_C19241A^</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode1_chromosomal_MDR_cyaA_consensus"/>chromosomal_MDR_cyaA</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode1_chromosomal_MDR_yidA_consensus"/>chromosomal_MDR_yidA</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode1_plasmid_MDR_AL513383.1_consensus"/>plasmid_MDR_AL513383.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode1_plasmid_MDR_CP029645.1_consensus"/>plasmid_MDR_CP029645.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode1_plasmid_MDR_CP029879.1_consensus"/>plasmid_MDR_CP029879.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode1_plasmid_MDR_CP029924.1_consensus"/>plasmid_MDR_CP029924.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode1_plasmid_MDR_CP029957.1_consensus"/>plasmid_MDR_CP029957.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode1_plasmid_MDR_LT904879.1_consensus"/>plasmid_MDR_LT904879.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode1_plasmid_MDR_LT904892.1_consensus"/>plasmid_MDR_LT904892.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
	<tbody>
<table>
* Amplicon has fewer than 51 target organism reads<br><p>(?) Prediction is based on amplicon with fewer than 50 reads</p>^ Using model that was trained without negative cases - results are likely to be less reliable<br/><br/><br/><br/><br/>
</div>
<br/><br/>
<div class=header_line><a name="barcode2">barcode2<br/>S. Typhi H58 + wastewater</a><div><a href="#Summary">Back to Summary</a></div></div>
<br/>
<table>
	<tbody>
<tr class=table_header><td>Amplicon  <img class="help-icon" title="Sample name, you can make it more informative by adding sample description using options -c and -d."></td><td>Tot. Reads  <img class="help-icon" title="Total number of reads that mapped to the amplicon regadless of length (i.e. -s and -l parameters)"></td><td>Rejected on Length  <img class="help-icon" title="Number of mapped reads that were not assesed by classifier due to not meeting -s and -l parameters."></td><td>% Target Org.  <img class="help-icon" title="Percentage of mapped reads that were classified as target organism and that satisfied -s and -l length parameteres"></td><td># Target Org.  <img class="help-icon" title="Number of reads that were classified as target organism and that satisfied -s and -l length parameteres"></td><td>All known SNPs  <img class="help-icon" title="Number of SNPs detected that match known SNPs defined in classifier model file you used.&#10Includes all alleles different from references that occur in >50 reads and no less than 0.05% of all reads"></td><td>Dominant known SNPs  <img class="help-icon" title="Shows those detected SNPs that match known SNPs in classifier model file you used and also are the most common nucleotide that position.&#10E.g. if at position 254 the reference is C, but in reads the most common nucleotide is T this would count in this column.&#10If T occcured, but the most common was C, this would not count in this column. "></td><td>Dominant unknown SNPs  <img class="help-icon" title="Shows those detected SNPs that DO NOT match known SNPs in classifier model file you used and also are the most common nucleotide that position.&#10E.g. if at position 315 the reference is A, but in reads the most common nucleotide is G this would count in this column.&#10If G occcured, but the most common was A, this would not count in this column. If position has three occuring nucleotides, only the most common is examined."></td><td>Implied Genotypes  <img class="help-icon" title="Genotypes implied by detected SNPs (but not reference alleles). Number in brackets indicates what % of target organism reads supports this genotype.&#10For high level genotypes, a range of possible gnotypes (e.g. 0-2 or 3-4) is reported."></td><td>Implied AMR  <img class="help-icon" title="AMR genotype implied by detected SNPs (but not reference alleles). Number in brackets indicates what % of target organism reads supports this genotype."></td></tr>

<tr class=alt_row><td><a href="#barcode2_1_3_4.3.1_consensus"/>1_3_4.3.1</td><td>143</td><td>65</td><td>100.00%</td><td>78</td><td>1</td><td>1</td><td>0</td><td>4.3.1 (96%)</td><td></td></tr>
<tr><td><a href="#barcode2_2.2.2_2_consensus"/>2.2.2_2</td><td>212</td><td>112</td><td>100.00%</td><td>100</td><td>0</td><td>0</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a href="#barcode2_2.3.1_v12_consensus"/>2.3.1_v12</td><td>191</td><td>50</td><td>100.00%</td><td>141</td><td>1</td><td>0</td><td>0</td><td>2.3.1 (18%)(?)</td><td></td></tr>
<tr><td><a href="#barcode2_2.3.2_consensus"/>2.3.2</td><td>555</td><td>103</td><td>100.00%</td><td>452</td><td>0</td><td>0</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a href="#barcode2_2.5_4.3.1.2.1_3.3_consensus"/>2.5_4.3.1.2.1_3.3</td><td>326</td><td>111</td><td>99.53%</td><td>214</td><td>0</td><td>0</td><td>0</td><td></td><td></td></tr>
<tr><td><a href="#barcode2_3.1.1_v5_consensus"/>3.1.1_v5</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode2_3.3.1_consensus"/>3.3.1</td><td>150</td><td>37</td><td>99.12%</td><td>112</td><td>0</td><td>0</td><td>0</td><td></td><td></td></tr>
<tr><td><a href="#barcode2_4.3.1.1_consensus"/>4.3.1.1</td><td>201</td><td>39</td><td>100.00%</td><td>162</td><td>1</td><td>1</td><td>0</td><td>4.3.1.1 (94%)</td><td></td></tr>
<tr class=alt_row><td><a href="#barcode2_4.3.1.1.P1_v2_consensus"/>4.3.1.1.P1_v2</td><td>72</td><td>13</td><td>100.00%</td><td>59</td><td>0</td><td>0</td><td>0</td><td></td><td></td></tr>
<tr><td><a href="#barcode2_4.3.1.2.1.1_4_consensus"/>4.3.1.2.1.1_4</td><td>433</td><td>186</td><td>99.19%</td><td>245</td><td>1</td><td>1</td><td>0</td><td>4 (99%)</td><td></td></tr>
<tr class=alt_row><td><a href="#barcode2_4.3.1.2_v3_consensus"/>4.3.1.2_v3</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode2_parC_v7_consensus"/>parC_v7</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode2_tviD_v2_consensus"/>tviD_v2</td><td>392</td><td>92</td><td>99.33%</td><td>298</td><td>1</td><td>1</td><td>0</td><td>known, but uninformative allele (95%)</td><td></td></tr>
<tr><td><a href="#barcode2_acrB_v2_consensus"/>acrB_v2</td><td>605</td><td>93</td><td>99.61%</td><td>510</td><td>0</td><td>0</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a href="#barcode2_gyrA_consensus"/>gyrA</td><td>157</td><td>64</td><td>100.00%</td><td>93</td><td>1</td><td>1</td><td>1</td><td></td><td>gyrA_S83T (99%)</td></tr>
<tr><td><a href="#barcode2_plasmid_ST6_C19241A_consensus"/>plasmid_ST6_C19241A^</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode2_chromosomal_MDR_cyaA_consensus"/>chromosomal_MDR_cyaA</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode2_chromosomal_MDR_yidA_consensus"/>chromosomal_MDR_yidA</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode2_plasmid_MDR_AL513383.1_consensus"/>plasmid_MDR_AL513383.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode2_plasmid_MDR_CP029645.1_consensus"/>plasmid_MDR_CP029645.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode2_plasmid_MDR_CP029879.1_consensus"/>plasmid_MDR_CP029879.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode2_plasmid_MDR_CP029924.1_consensus"/>plasmid_MDR_CP029924.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode2_plasmid_MDR_CP029957.1_consensus"/>plasmid_MDR_CP029957.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr><td><a href="#barcode2_plasmid_MDR_LT904879.1_consensus"/>plasmid_MDR_LT904879.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
<tr class=alt_row><td><a href="#barcode2_plasmid_MDR_LT904892.1_consensus"/>plasmid_MDR_LT904892.1</td><td>*</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td><td>-</td></tr>
	<tbody>
<table>
* Amplicon has fewer than 51 target organism reads<br><p>(?) Prediction is based on amplicon with fewer than 50 reads</p>^ Using model that was trained without negative cases - results are likely to be less reliable<br/><br/><br/><br/><br/>
</div>
<div class=header_line>Consensus sequences</div>
<br/>
<table>
	<tbody>
<tr class=table_header><td>Sample  <img class="help-icon" title="Sample name, you can make it more informative by adding sample description using options -c and -d."></td><td>Amplicon  <img class="help-icon" title="Amplicon name"></td><td># of target reads  <img class="help-icon" title="Number of reads classified as target organism"></td><td>Consensus <br/> Mismatches <br/> vs Reference  <img class="help-icon" title="List of position where most common nucleotide does not match the reference sequence. If allele has known consequence this is also listed.&#10This list usually doesn't match the one in Sample's table because this column only reports cases where most common nucleotide doesn't match reference."></td><td>Consensus sequence  <img class="help-icon" title="Amplicon consensus sequence. If the most common nucleotide is deletion this is reported as '-'. Insertions are not currently supported."></td></tr>
<tr class=alt_row><td><a name="barcode1_1_3_4.3.1_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>1_3_4.3.1</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode1_2.2.2_2_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>2.2.2_2</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a name="barcode1_2.3.1_v12_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>2.3.1_v12</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode1_2.3.2_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>2.3.2</td><td>1762</td><td></td><td>ACGATAAACCGCTTCCGTCACCATCCCCGCCGGGTTAGTAAAGTTAATGACCCACGCATTCGGACATAATTCTTCAACATCCTTAATGATGTCAAAAATAACCGGAATGGTACGCAACCCTTTAAATAAACCGCCAGCGCCGTTGGTTTCTTGCCCCAGATAGCCGTGGCTAAGCGGGATACGCTCGTCCAGCTCACGGGCTTTGAGTTGACCAACGCGTAGCTGGGTAGTGACAAAATTAGCGCCTTTCAGCGCTTCCCGGCGATCCAGCGTTTTATACAATTTTAGCGGAACGCCTGCTTTATCAATCATTCGCTGGCAGAGATCATAAATAATGCCCAGCTTCTTTTTCCCGTCTTCAACGTCAACCAGCCATAATTCGGTGACAGGTAATTCATGGTAGCGCTTAATAAAGCCTTCAAGTAATTCAGGGGTGTAGCTGCTCCCGCCACCAATAGTGACGACTTTTAATTTCTGGCTCATAATATCTCCCTTCAGTGTATGACACTGATATGTATTACGCTATGCCCGCTCGTAAGAAATAGCGTGACGTTCCCCGCGCTATTGGGGCAAGCGTGAAATGTAACTAATCGAGCAATGTAAACCGTTACTCATTAATTTCCGTCAATCGCTTCCGATAACTATTCGGCGTGAATGAGGTCATCTTCTTAAACGTTTTAATAAACAGACTTGGACTACTGTACCCGGC</td></tr>
<tr class=alt_row><td><a name="barcode1_2.5_4.3.1.2.1_3.3_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>2.5_4.3.1.2.1_3.3</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode1_3.1.1_v5_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>3.1.1_v5</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a name="barcode1_3.3.1_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>3.3.1</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode1_4.3.1.1_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>4.3.1.1</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a name="barcode1_4.3.1.1.P1_v2_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>4.3.1.1.P1_v2</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode1_4.3.1.2.1.1_4_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>4.3.1.2.1.1_4</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a name="barcode1_4.3.1.2_v3_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>4.3.1.2_v3</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode1_parC_v7_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>parC_v7</td><td>16</td><td></td><td>TGGCACAATCACTAAACGCGTCGGGTTTTCGTGATCCGATTCATCGCGCAGGTCGTCCACCATCGGCAGTTTTTTATTGCGCATCTGCGCAGCAATCTGCTCCAGCACTTTTGCGCCAGATACCTGATGCGGTAACGCGGTGATCACCACGGCGCCGTCTTCTTTGGTCCATACCGCGCGCATACGCACGGAGCCACGCCCATTTTCGTAAATTTTACGAATTTCCGCACGAGGGATAATGATCTCCGCTTCGGTCGGGTAATCCGGCCCCTGGACGATATCCAGCAACTGATCCAGCGTCGTTTTCGGCTGCTCAATCAGCGTAATCGCCGCTTTCGCGACTTCGCGCAGGTTATGCGGCGGAATATCTGTTGCCATGCCCACCGCAATACCGGTGGTGCCGTTCAGCAGGATGTTCGGCAGACGCGCCGGTAACATTTTCGGTTCCTGCATGGTGCCGTCGAAGTTTGGCACCCAGTCCGCAGTCCCCTGACCGAGTTCGCTTAACAGCAGCTCGGCGTATTTGGACAGGCGGGATTCGGTATAACGCGTCGCCGCGAATGACTTCGGATCATCCGGCGCGCCCCAATTCCCCTGGCCATCGACCAGCGGGTAACGGTAAGAGAACGGCTGCGCCATCAGCACCATGGCTTCATAGCAGGCGCTGTCGCCATGCGGGTGATACTTACCCAGTACGTCG</td></tr>
<tr class=alt_row><td><a name="barcode1_tviD_v2_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>tviD_v2</td><td>26</td><td></td><td>GCAAGCTGCTTAGTGATCGATAAGGTAATCTCACGTATTTGGTCCTGTTTCTCTTGTCGCCCCTCATCAAGATGATGATTAGTATTATTGATCACTCCCTTAAAAATAGCCGCAACGATCGCGTCACCAAACATATCGGTATAATGAGTAAGATCGTCGCCATTTTTTTCCAGGGCTCGTTTCTGCCCCCACTTTTCCATCAAATTTGTCGGGTTAACGCAGGGAATATCCATCCTGGCGGCGATGGTGTCGACATTTTTAATCAACCGACTACGGGATAGAATGACGGTTCCAGCACGAGTCACGGCATCAACATGTGTCATAAACATAACGCGATCGCGACCAAGAAGCTGAACGATTTCCATCATATCCTGTTCGATAGCATGCTCATCCATCTGCTCGACATTGATATTGCTTAATAAGGCACGATCGTCTTCCGACATTCCGGCAAAACAGGGATCGTCTTTGAGATAGGCCAGCAGTTTGTGTCGATTTCCTTGCGTGGCCAACGACCAGTACATGCACGCCCGCTCTGTTTGGCTAAAAAATTCATAGAAATGACGAGTTAAATAATTTATTTGTAAGCAGAACCCATAGGCCATAATCTTCTTCTGACTTGAAATCTCAATGATATAGAAATCAGAACGACTATGTACGTCGGTGTTTGAATAGTTTACATTAGGTCTAAAAATAAATGGTCGGACCTTTTCAGGAATGTCTATCAAGCCCAAAATAAACCGAATCTGCTGTAATGCCTCGCTGCTCGTATGCGTAAAACCATATATTCTCTTAAAGTTTGCCTGAAAATTAAAATAAGGC</td></tr>
<tr><td><a name="barcode1_acrB_v2_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>acrB_v2</td><td>1150</td><td></td><td>CACAGGAAGACGACGATTAGCGAGATAGCATACAGGGCAGGGGCCTGGTTGCCGGACAACCGCTCCTGGTAGGACATCCCGGTCCAGTCATACCCAATGCCTGACGGCAGCTTGCTGGCCAGTTCTTCCATCATCGCCATCGCTTCACCAGTACTCTTGCCTGGCGCCGCCTGACCCAGAATTTCCATCGAAGGCAGACCGTTATAGCGTTCCAGACGCGGCGAACCATATTCCCAGCGGGAAGAGGAGAATGCGGAGAACGGCACCATCTGACCATCGCTACCACGAACGTACCAGTCGTTAATATCATCCGGCAACATGCGGTATTTCGCTTCGGACATCACGTAAACTTTCTTCACACGACCGCGATCGATAAAGTCGTTTACATAGCTGCCGCCCCATGCTGCGCCCAGCGTGGTATTAATGTCGCTAATAGACACGCCCAGCGCCTGAGCTTTTTCCTGGTCGATATCGATTTTAAACTGCGGCGTATCTTCCAGACCGTTAGGTCGAACGCCGACCAACAGATCAGGATATTTCGCCACCTCGCCGAACAACTGATTACGTGCCTGGGTGAGTTTTTCGTGACCAAGTCCCGCCTGGTCAATCAACTCGAAGTCAAACCCGGTTGCGGTGCCCAGTTCAACAATCGCCGGTAGGTTAAAGGCGAAGACCATCGCATCTTTAATTTGTGAAAACGCTGCGGTTGCCCGCTGGGTAATCGCTTCAACCTTGTTTTTTTCGCCTGGACGATCGGCCCAGTCTTTCAACGACACAAATGCAATACCGGTATTCTGACCGCGCCCTGCAAAACCGAAGCCGTTGACGGCGAATACCGATTCAACGTTGGCTTTCTCTTTGTTCAGATAGTAATCCGTGACCTCATCCAGCACTT</td></tr>
<tr class=alt_row><td><a name="barcode1_gyrA_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>gyrA</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode1_plasmid_ST6_C19241A_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>plasmid_ST6_C19241A</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a name="barcode1_chromosomal_MDR_cyaA_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>chromosomal_MDR_cyaA</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode1_chromosomal_MDR_yidA_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>chromosomal_MDR_yidA</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a name="barcode1_plasmid_MDR_AL513383.1_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>plasmid_MDR_AL513383.1</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode1_plasmid_MDR_CP029645.1_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>plasmid_MDR_CP029645.1</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a name="barcode1_plasmid_MDR_CP029879.1_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>plasmid_MDR_CP029879.1</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode1_plasmid_MDR_CP029924.1_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>plasmid_MDR_CP029924.1</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a name="barcode1_plasmid_MDR_CP029957.1_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>plasmid_MDR_CP029957.1</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode1_plasmid_MDR_LT904879.1_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>plasmid_MDR_LT904879.1</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a name="barcode1_plasmid_MDR_LT904892.1_consensus" href="#barcode1">barcode1<br/>S. Typhi negative wastewater</a></td><td>plasmid_MDR_LT904892.1</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode2_1_3_4.3.1_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>1_3_4.3.1</td><td>78</td><td>Pos:1655 (96%) , Change:A>G, Implication: 4.3.1</td><td>CGATGGTACTGAACAACCCTTTCGCATGCTTTTCACCGACGAGCGGTACCAGCGTGGAGCTGGGGCGGATGACATCCATCTTGCGTACAAAGAAGAAGTAGGCGAAGGCCACGGAAAGGAAGCTATAGGTTGCCCACGGTAGCGGCCCCCAGTGGAACAGGCTGTACGCCAGGCCTATCTCTTTCGCGCCCGTGGAATTGGGTTCCAGGCCGAACGGCGGGGTGGAGATATAGTAGTAGATCTCAATAGAGCCCCAGAACAGAACCGCAGCGGATGTACAGGAGGCGAACATCATGAAAATCCAGCTGGCGGTGCTAAACTCCGGTTTTTCGTCACCTAATCTCTTTTTGGCATAGGGGCCAAAAACCAGCCAGAACCAGCTGAAGAGCATGACGACCATATACCATTCAAATGCCCAACCCCAGACGTTGGTGACATAACTGAATACCGCATTAATAACGACGTTAGCGGCATCAAGATCTCTGACTGTTAGCCAGCAAAGTATCCCGACAATTATTAATAGCGGAAAAAAAACCTTCGGTTCAATTCCCGATTTTCTCTTTTCATTTTTCATAAGTGAATTCCAGTGTGAAAACGAAATTTATTTAGCGTGCCAGTTCATGAACGTTTGTTTATTTATTATTAATGATTTGTTGATAAGTTTTTAACCTCTGGGACGCCTCAATGAGTAACTCATAATGGGTATTCATTTCGGGTGATTCTGTTATCTGTGTCACACTTTTTTGCGTTGATTATTTTGTGTTTAAATTTTGATTTTTTCTTTATTTATCAATCTGTTGGTTGTGGTTTTTTAACTTTTTTAAGTATCTGGCGTTCAATATTGTTGAGCCAGCAAACAATATTGAAAATTTTCATATTTGGCATATTTCTTTCAATATTGGTGAGGTGCGTAACACTATTGAAAGTTACGAATATCAATGTGTGACATTTTCAATATTGGTGATTAAGGTTTTATTTCCGAATTAAAGAGCGTGATATCTGTATTTAACACCGCCGATATAAATACGTTTCCTTCATGATTTCTGGAGATGCAATGAAGATAATTACTTGCTACAAATGCGTACCTGATGAGCAGGATATTGCGATTAATAATGCTGACGGAACACTCGATTTCAGCAAAGCGGATAGCAAAATCAGCCAATATGATCTGAATGCAATTGAAGCTGCTTGCCAGTTAAAACAGCAGTTGGGAGACGCTCAGGTTGTCGCCATGAGCGTTGGCGGCAAAGCGCTGACTAATGCAAAAGGGCGCAAAGATGTCCTCTCTCGTGGCCCTGATGAACTGATTGTGGTCATTGACGATCAATTTGAGCAGGCGCTGCCGCAACACACCGCTACCGCGCTGGCCGCCGCGGCGCAAAAATCAGGTTTCGATCTACTTATCTGCGGCGACGGTTCTTCCGATCTTTATGCTCAGCAGGTTGGTTTGCTGGTGGGCGAAGCGCTGAACATTCCGGCGATTAACGGCGTGAGTAAAATCCTCTCTCTTACCGACAGCACATTGACGGTAGAACGCGAACTGGAAGATGAAGTTGAAACGCTGAGCATCCCGCTCCCGGCGGTGATCGCGGTTTCCACCGACATTAACACCCCACAAATCCCTTCGATGAAAGCCATTCTTGGCGCGGCGAAAAGACCGGTTCAGGTCTGGTCGCCTGCGGATATCGGGCTGAACAGCGT</td></tr>
<tr class=alt_row><td><a name="barcode2_2.2.2_2_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>2.2.2_2</td><td>100</td><td></td><td>GCACAGTTCATCCGAGTGATTAACAATCCACCACGCCTCTTCGTGGCGAACAAGACTGAGCGCGATATCGTCTTTCCCATGTTGCTCAAAAAGCGGAATGTAGGGGGCGGCGGCAGTAAATACGATGTATGTCCCTGCGGGATACTGTGCGTTATCCGTTATACCCTGCGACTGTATTTTCCGTAATTCCCACATGTCTTGTTTCATTATGTCCCTTAAATGTATTTATTTTTCTGGAGGAAACGTTTAGGGAGTTTTAATTCATAAATAATTTTAACTAAATTCATAGGGAGTCATTATTGATGACACCCTTTTTATTATTTTCGTCGTGAATGCATTGTTGTATGCATAGATGTCTTTTTTGAAATATTATTTCTTTTAATTCTGTAATTGCGATTTTTATCTTGTTGTTCATTGATTATTTAATATTAACGGCTTGGTTGCTAATTAGTCCCTGATCTGTATCATTGTTTTGTTTCGATATTTTGTCGAACATTTTAATAAAGAATTGTTTATATATTCATGTGCATTGATGCATTATTTTTATGAATTTTTATGTCACAAGGCATAACAAATGGAAACTCCTGTTTCACGCAGTGCGTTATATGGAAAACTGGCCGGCCCACTATTCCGGTCGCTGGAATCGGCAACGGCATTTTGCAAACTACGCTCTAATCCCTGGGGTGAGCTGACTCACTGGCTGCACCAGTTAACACAGCAGCCCGATAACGATATTCTCCACGTTCTTCGGCATTACCAGATCCCTCTTTCTGATGTGGAGAAAGCGTTACTCCGGCAACTGGATATGCTGCCCGCCGGGGCCAGCGCCATTAGTGATTTTTCTCACCATATCGATCTCAGCGTTGAAAAGGCCTGGATGCTGGCGAGCGTCCGTTACGGCGATAACAAAATTCGCAGCGGCTGGTTGCTGCTGGCCTTGTTGACCACGCCAGAACTGCGTCGGGTACTGAGCAGTATCTGCGCGCCGCTGGCCACGCTTCCGGTTGATGAACTGACGGAAATACTGCCCTCGTTGATCGAAACATCGCCGGAAGCGCAGGAGCGCCCTTACGACGGCTCCGGTCTGGCATCAGCCATTCCCGGTGAAAGCAGCCAGGCGATTCCCAACGGCGTGCAGGACGGTAAATCCGCGCTGGCAAAATACTGTCAGGACATGACGGCACAGGCGCGCGACGGCAAAATCGACCCGGTGACGGGGCGTGAGCATGAAATCCGCACCATGACGGATATTCTGCTGCGCCGTCGCCAGAATAATCCCCTACTGACTGGTGAGGCGGGCGTCGGAAAGACGGCGGTCGTCGAAGGTTTTGCCCTCGCGATTGCGCAGGGGGAAGTGCCGCCCGCGCTGCGGGAAGTACGGCTGCTGGCGCTGGACGTTGGCGCTCTGTTGGCCGGAGCCAGCATGAAAGGCGAGTTTGAATCGCGTCTGAAAGGGTTACTGGAAGAGGCCGGGCGCTCGCCGCAGCCGGTTATTCTGTTTGTCGATGAAGTTCACACTCTGGTGGGCGCGGGCGGCGCATCCGGCACGGGCGATGCCGCTAACCTGCTGAAACCGGCGCTGGCGTGCGGCACCCTGCGGACTATCGGCGCCACCACCTGGAGCGAATACAAGCGCCATATTGAGAAAGATCCGGCGCTGACCCGTCGTTTTCAGGTGTTGCAGATTGCCGAGCCGGAAGAGATCCCCGCAATGGAAATGGTGCGTGGTCTGGTGGATACGCTGGAAAAACACCATAACGTACTGATTCTGGATGAGGCGGTACGTGCGGCGGTACAGCTTTCTCACCGCTACATTCCCGCCCGGCAGTTGCCGGATAAGGCCATCAGCCTACTGGATACCGCCGCGGTCCGCGTGGCGCTGACGCTGCACACGCCGCCTGCCAGCGTACAGTTCCTGCGTCAGCAGCTAAAAGCGGCGGAAATGGAACGGTCGCTGTTGCAGAAGCAGGAAAAAATGGGGATTCAGTCAGATGAGCGGTGCGATGCGCTGACGGCGCGAATTTTCTCGCTCAACGATGAACTGACTGCATCCGAATCCCGCTGGCAGCGGGAGCTGGAACTGGTACATACGTTGCAGGAACTGCGTGTCGCAGAGTCTGATGC</td></tr>
<tr><td><a name="barcode2_2.3.1_v12_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>2.3.1_v12</td><td>141</td><td></td><td>TTCAATAACGAAGCGATCTCCTGCAGTATTTCAGAACGTCCAACTATGCCGATGATCGATGACATGCTCATGAAGGCATTTGCCAGACTTGATGCGGGTACCAATCCGGTTCTTCATTCAGATCAGGGCTGGCAATACCGTCATTGATGGTACCAGTATCAGTTACGAGAGTTTGGCATCATCCAAAGTATGTCCCGCAAAGGCAACTGCCTTGATAATGCCTGTGCAGAATTTTTTTTCGGCACTTTAAAATCAGAAAGTTTTTACACCAGTAAATTTAAGGATATTGATGAACTTAAGATAGCTATTGAGGATTACATACGATACTACAATACCCGACGAATTAGCCTTAGATTTAACGGACTCAGCCCGGTTGAATACCGCCTAAAGAGCTATCCAGGCAGAAATTAACAAGGTCTGTACAGGGATGCTATCAGGAAACCTGAACGTTGGTGGATACGCTCACGTAAATGTCTGCTATGAGCGAGGAGCGGACGCCCTATTTCACAAAAATTAGTATTGTTCTACAAGTAACTCTCACAATGAGGTTTTATGAAAATCAGAAGGTTCAGTAATGGGGATGAAATATCGTTATTCAGAGTTTTTTTCTCATCTGTACATACGATCGCATCACACTACTACACACGTGAACAAATTGATGCTTGGGCCCCAGCAGATATCGACCTGGAGCGATGGGCCAATCACATTAAGGAATTACAACCTTTTGTTGTAGAGCTGGATGGTGAAATAGCTGGATATGCAGATGTTCAACCAAATGGATACATTGATCATTTCTTTGTATCAGGAACCTATTCA</td></tr>
<tr class=alt_row><td><a name="barcode2_2.3.2_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>2.3.2</td><td>452</td><td></td><td>ACGATAAACCGCTTCCGTCACCATCCCCGCCGGGTTAGTAAAGTTAATGACCCACGCATTCGGACATAATTCTTCAACATCCTTAATGATGTCAAAAATAACCGGAATGGTACGCAACCCTTTAAATAAACCGCCAGCGCCGTTGGTTTCTTGCCCCAGATAGCCGTGGCTAAGCGGGATACGCTCGTCCAGCTCACGGGCTTTGAGTTGACCAACGCGTAGCTGGGTAGTGACAAAATTAGCGCCTTTCAGCGCTTCCCGGCGATCCAGCGTTTTATACAATTTTAGCGGAACGCCTGCTTTATCAATCATTCGCTGGCAGAGATCATAAATAATGCCCAGCTTCTTTTTCCCGTCTTCAACGTCAACCAGCCATAATTCGGTGACAGGTAATTCATGGTAGCGCTTAATAAAGCCTTCAAGTAATTCAGGGGTGTAGCTGCTCCCGCCACCAATAGTGACGACTTTTAATTTCTGGCTCATAATATCTCCCTTCAGTGTATGACACTGATATGTATTACGCTATGCCCGCTCGTAAGAAATAGCGTGACGTTCCCCGCGCTATTGGGGCAAGCGTGAAATGTAACTAATCGAGCAATGTAAACCGTTACTCATTAATTTCCGTCAATCGCTTCCGATAACTATTCGGCGTGAATGAGGTCATCTTCTTAAACGTTTTAATAAACAGACTTGGACTACTGTACCCGGC</td></tr>
<tr><td><a name="barcode2_2.5_4.3.1.2.1_3.3_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>2.5_4.3.1.2.1_3.3</td><td>214</td><td></td><td>GGTTCGTTGTCCATTTCGGTATGACCAGGACGAGAAAATTTTATGACTGCGTAAAGATAATCTAAGTAAATATTTAGCATCAATGGAACATGCATGGTGCTGATTATTAAAACCACAGTGTTCAGGTAAATAAATATGGCTACCGGTCTCTAAATCAACGTCCCGAAAAAAATGAATCAAATAATCGGATAAGTCAAATCTGATGTTATTTTTCATGGGACGCCCTCTTTCAAACAGTCTCTTTTTTGCATTCCTTTAAAACCAGCACCATCATTTTATATAAAAATCATCACGAAGTATGCTTCTTTTAACGATGACATCAAATCCTCCCTCCTTTTGCATCAACTTACGTATCCCTGAAATGGCGAGACCAGGCTAAATCCACTTGAGGTTGCTCACTGAAAACCTCATCCCGGAACAAACTCAACCGCGCTTCCCCGCCACGGCCCTTTCGCCGAAAAACCGATAATGGGTATAAATATACCGCTATAAGCTATACCCACACACTTCACATTGTTTCTTATCGTTAATCCGATACGCTTTTTTCACATTATTTCTGGCGGTATACGTCGGTGCAAAGTGAATCGGGCTCGTTTCCGCGCTCGCGCGTAACACAATGATTTACCTTGCTGGCGGAACGTCAACGGACCCGCAAAAGCGCCTTATGCCCCTGGCGGAGCGTGGTGCAAAAAGGAGAAAAAATACGCGTTATGAATAACGAGGAGACCTTTTATCAAGCCATGCGTCGTAAGGGAGTGACCCGACGCAGCTTTCTCAAATTCTGTAGCCTTGCCGCCACATCGCTGGGACTGGGCGCCGGAATGACGCCAAAGATCGCCTGGGCGCTGGAGAATAAACCGCGGATTCCGGTGGTCTGGATTCATGGACTGGAATGCACCTGCTGTACCGAATCCTTTATCCGTTCCTCGCACCCGCTAGCCAAAGATGTGATCCTCTCGCTGATTTCCCTCGATTATGACGACACCCTGATGGCCGCCGCCGGCGCACAGGCCGAAGAAGTCTTTGACGATATTACCACTCGCTACGCCGGGAAATACATTCTGGCGGTGGAAGGCAATCCGCCGTTAGGAGAGCAAGGAATGTTCTGTATCAGCGGCGGCCGCCCGTTTATTGAAAAACTGAAGAAAGCCGC</td></tr>
<tr class=alt_row><td><a name="barcode2_3.1.1_v5_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>3.1.1_v5</td><td>43</td><td></td><td>ATGGCTTCTGGTTGGCTTTGACCAACTGGCAATGCAGGCTTTTCAGGCCGCGCATTATGGTTTTATCTCCCGGCAAGAGCTTCACCGTAGCCTTCGACGGGGAGGCCATCTAATCCGTCAGATCTATGGTGCCGCTCAGAAATATCGTTTTTTTCAGGTGAACCGCCGCGATTTTGCATTACAGAATGCTCAGTACCATGAGGCAATCAGGAACGCAGGTGAAATTGACGAAGCTGTTTTGTTGGGGCAAAAACGTTCCTCTTTTTCACCCCCAGTCAGTAAGGAGAGTATTGAACTTCTACTGGCGGCAAATGCAAAAGCAGACAAAGCAGATATTGTGCAATTACTGTAAGAACCGTTAGAAGAACAGATTGTGGCTGACCCTTCTTTATTTGTCGCTGATTCGAAACGCATCGATGGTTAAGCCCGCAGAAAGGCTGGTTTTCTGAAAGGAGTGTTATCATGCGTCTGGTGTTGTGTGAAAAACCTTCTCAGGGCAGAGATATTGCGAAGTTTCTTGGCGCCACTCAAAGAGGCGAAGGATTCCTGAATGGTCCGGGAGTTACAGTCACCTGGGCTCGCGGCCATCTGCTGGAGACAGCTGAACCGGAAGTATATGGCGAACAGTATGGTAAGCCATGGCGCACCGATGTGCTTCCTTTATTACCACAGCAGTGGAAACTGGTAGTCAAAGCGGATGCAAAAGCCCAATTCGCGGTAATTAACCGTCTGCTTAAACAGGTAGATGAGGTGGTAATTGCCACGGATGCTGACCGGGAAGGTGAAGTAATCGCCCGTGAGTTGCTGGATTACTGCAAGTTTCAGGGGCGTGTATTCCGGTTGTGGATGTCAGCTCTGGATGATGCCAGTATCCGGTCAGCGTTAGCTGATTTGTGGCCGTCTTCCCGAACTGAATCGCTTTATTATGCTGGCGTGGGCAGGAGCAGGGCTGACTGGCTGATCGGTATGAATCTGACTCGCCTGTTTACACTCCTGGGGCGTGAAGCGGGTATCGGTAACGTACTTTCAGTTGGCCGGGTACAAACGCCAACGCTGGCGATAGTGGTAAAGCGTGATCGTGAAATTGATAACTTCGTGTCGAAACCGTACTTCGAAGTTATTGCCACGATGGCCGTGAATGGCGTGTTGTT</td></tr>
<tr><td><a name="barcode2_3.3.1_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>3.3.1</td><td>112</td><td></td><td>TCCGTTTTGCGAAATCGTTCCCGGAAAAATGATCCATTTCTGTCACACTCAGAACGATTTGATAACAACAAGAGGTCATAGGGATGAAAAAATCATTACTCGCTGTTGCTGTGGCAGGGGCTGTTTTGTTGTCATCCGCCGTACAGGCGCAGACAACGCCGGAAGGTTATCAATTACAACAGGTGCTGATGATGAGCCGCCATAATCTGCGGGCGCCGCTGGCGAATAATGGCAACGTACTGGCGCAGTCGACGCCGAACGCCTGGCCGGCGTGGGACGTTCCCGGCGGGCAACTGACGACGAAAGGCGGCGTGCTGGAAGTCTATATGGGACACTACACACGTGAATGGCTGGTCGCGCAGGGGCTGATACCGTCGGGAGAATGTCCGGCGCCCGACACGGTATATGCCTATGCGAATAGTTTGCAGCGCACCGTCGCCACCGCGCAATTTTTCATTACCAGCGCTTTCCCCGGCTGTGATATTCCTGTTCATCATCAGGAAAAAATGGGCACTATGGACCCTACCTTCAATCCGGTGATTACCGATGATTCCGCCGCGTTCCGGCAACAGGCCGTACAGGCGATGGAAAAGGCGCGTAGTCAGCTACATCTTGATGAGAGTTATAAACTGCTTGAGCAGATAACGCATTATCAGGACTCGCCGTCCTGCAAAGAGAAGCATCAGTGTTCGCTAATCGACGCGAAAGATACCTTCAGCGCGAACTATCAGCAAGAGCCTGGCGTGCAGGGGCCGCTGAAAGTAGGGAACTCGCTGGTGGATGCGTTTACCCTGCAATATTACGAAGGCTTTCCGATGGATCAGGTCGCTTGGGGCGGGATCCACACCGATCGGCAGTGGAAGGTGCTGTCAAAACTGAAAAACGGCTACCAGGACAGCCTGTTTACCTCACCCACGGTGGCGCGCAATGTCGCTGCGCCGCTGGTAAAATATATCGATAAGGTGCTGGTTGCCGAGCGCGTTAGCGCGCCGAAGGTTACCGTGCTGGTGGGGCATGATTCCAATATCGCGTCGCTGCTGACGGCGCTGGATTTTAAACCCTATCAGCTCCATGACCAGTATGAGAGAACGCCGATTGGTGGTCAGCTTGTCTTCCAACGCTGGCATGACGGTAACGCTAACCGGGATTTGATGAAAATCGAGTATGTCTACCAGAGCGCC</td></tr>
<tr class=alt_row><td><a name="barcode2_4.3.1.1_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>4.3.1.1</td><td>162</td><td>Pos:456 (94%) , Change:T>C, Implication: 4.3.1.1</td><td>CTGGCCTGATACCTGGATGTATACCGCCAACACCAGCGATGTCGCTATTCTCAAGCGAGCGCACCAAAAGATGGTGAAAGCTGTCGATACTGTCTGGAAAGGTCGGGCCGAGGGGCTGCCTTATAAAGATCAGAACCAACTGGTGACAATGGCCTCGATTATTGAAAAAGAGACGGCTGTCGCCAGCGAACGCGATCAGGTGGCCTCAGTCTTTATTAATCGCCTGAGAATCGGTATGCGCCTTCAGACCGATCCCACCGTGATTTACGGGATGGGGACGAGTTATAATGGTAACTTGTCGCGTGCGGATCTGGAAAAGCCGACGGCTTATAACACGTATACCATAACCGGGCTGCCGCCAGGACCGATTGCATCGCCCAGCGAAGCGTCATTGCAGGCGGCGGCGCATCCGGCGAAAACGCCGTATCTCTATTTTGTGGCCGACGGTAAAGGCGGCCACACATTTAACACCAATCTTGCCAGCCATAATCGGTCAGTGCAGGAGAAAGTGCTTAAGGAAAAAAATGGGCAGTAATTATATCGTCATCGAGGGCCTGGAAGGCGCCGGAAAAACCACTGCGCGCGACGTGGTGGTGGAGACCCTTGAGCAACTGGGTATTCGTAACATGATTTTTACCCGTGAGCCGGGCGGTACGCAGCTTGCCGAAAAACTAAGAAGTCTGGTGCTGGATATCCGATC</td></tr>
<tr><td><a name="barcode2_4.3.1.1.P1_v2_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>4.3.1.1.P1_v2</td><td>59</td><td></td><td>TAGGTCGACCAGCGCAAATTCATGACTGTATAAGGCTTTCGCCATATCCGGTTTCACAAACATATTATGCCAGTTAAGGCTATAGGGCCACGGGCTTTCCGGACCGTGATAAAAGAGGATTGGTACCACCAGCGGCAAGGTATCATGGCCGGCATCCAGATGACGCTGCATGGCGGCAATGGCGTACCGCATCAGGCGAAAAGCCATCAGCCGGTCTGATTTGCTCTGATGTTCAATAAGCGCGTAAACATACCCGGGGCCGTGTTGGGTATGCAGAGAGTAGAGAATATCTGAATAGCGGCTGCGTAAATTGTCTTCAACAAAACTCCCCGACTCCAGCTTCAGTGAATCCAGATCGCAGATTGCAAGAAAGTCTTCTGGCAAGTGGATAGCCAGAAAGTCGCGGGCGACCTCTTTTTGCATCAGCATCTGTCGGAATACCGCATCGTGCGGCGTGGTGGGTAGACGTTCCATGCCCCGTCCTTTGTTGCCTGACTGTAGCGACCACTTTGTCAGCGACAACGAGGACAAACCAGGGGGCTATTACTCTGTTTGCGAGTCGTCTTCCGTACTTTTTAACTTGTTTTCATAATCACGCATAACTATTTCCAGCGCGCGTAGCACCGTTCGTGCCGGTAAGCGATTCTCTTCAAGCAGCACAATTAAATCCACGGCCAGTTTGACTTCATCAGGGGCATTTTCAAGCGACATACTTTCTCCTGGGTTAACGGG</td></tr>
<tr class=alt_row><td><a name="barcode2_4.3.1.2.1.1_4_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>4.3.1.2.1.1_4</td><td>245</td><td>Pos:1047 (99%) , Change:G>A, Implication: 4</td><td>CAGGCCTGGTTTGACAATCTGTTTGTCCGTCAGGATCTCGGCAATCAGTATTATCTCCAGGCTGGTCGGATGGATCAGCGGAATCTGTCCAGCGCCACGGGGGGGGATTTTGGGTTCAGTCTGCTTCCCCTGAGCCGGTTTGATGGATTACGAACCGGGACCACCCAAGCTTATGTTAACCATGAGGTGGACCATAATGCCACTCCGGTTATGGTTCAGGTTACCCGAAATGCCCGTATTGATATTTATCGTGGCAGCGAGTTGCTGGGGAGTCAGTTCCTGACCCCGGGAATGCATACCCTGGATACTCATTCTCTTCCACCGGGAAGCTATCCTCTGGCGTTGCGGGTGTATGAGGATGGGATTCTGCGGCGAACGGAGACCCAGCCCTTCAGTAAGGGGGGCAATAGCTTCAGTGCACAGACCCAGTGGTTTATTCAGGGCGGGCTGGAAGATACCGGGGATAAAGCCAGCCATTATGACGGTGAGACTGTCATGGCTGCCGGATTCCAAACTGGGCTGCGGAAAAATATCAGTCTGACCGAAGGTATCTCTCTGGCACATGAGGCCTGGTACAGTGAAACCCGACTGAATTCACAGCATGCAGTGCTGGATGGCACGCTGGACCTTTCTGCCGGGATACTGCATGGGACAGACAGCACGAGCGGTAACACTGAGCAGGTGACATACAACGACGGATTTTCCGCGAGTCTGTGGCGTAACCATACGGAAAGTGATGCCTGTAGTGGTCGTCATCCACAGTCAGTGCATGCCAGTATGACCTGCCAGACTTCGATGAACGCCTCCCTGTCGGTTTCGGTGGGGAACTGGTATGCCCTACTGGGATACAGTACCAGCAGGACAGAAGGTCGGCCGGTTTACCGGGGATATGATGATAACAGTGACAAAGAAAATGTGTTCTGGCGACAGGCATACATCCCTGCCTCTCACCGCGAATCTGCTCAGGCTAGTGCAACGTACAGCCTTAATATGGCTGGCATGAATATTAATACCCATGGGGGAGTATGGCGAACCCGAAATGACGGAATGAATGATGATGGCTTGTTTATGAGTGTCAGTGTGTCATATGCCTCTCAACCACCGACAATGACTGGCAGTAATAGGTATACCTCAGCCGGGACCGATATTCACAGTAGCCGGAATCAAAAAACACAGACGTCCTGGAATGTGAACCATGTGAGATCCTGGCAGCAGGATCTGTATCGTGAACTGTCGGTGGGTTTCTCCGGTTATAACGACGACAGCTGGAGCGGGAGTCTCGGCGGACGCATGAGCGGCCGTATGGGTGAACTGAGCGCCACTATCAGTAACTCCCATCAACGTAATGCGGGCAGCGCCAGTTCACTCACCGCTGGCTACAGCTCGTCTCTGGCGTTATCCCGTAATGGACTGTTCTGGGGAGGTGGTCAGGACGGTGAACCGGCCTCTGGCATGGCGGTGAACGTGGAGTCAGAGGGGGACGAGGGCAGTAGCGGGAAAGTAGTCAGCGTTCGTGGCAGCAGCCAGCCGTTCAGTCTCGGTTTTGGTCAGCAGTCGCTGTTGCTGATGGAAGGCTATAACGCCACGGAGGTGACCATTGAGGATGCAGGGGTTAGTTCACAG</td></tr>
<tr><td><a name="barcode2_4.3.1.2_v3_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>4.3.1.2_v3</td><td>8</td><td></td><td>TGTTTCTGGCTTCGCTGCTGGGGGCTGCGCTGGACTTTCGCCAGATTGCGTTTCCGGCGTCTGAGAAGGCGCTGGCGTCGCAGGCTGACTGCTGGCGGCCGGCGTCGCCTCGGTTTTAACATTATTGTCATCACATCCTGTTGCGCCCAGGGTCATTAGCGCAATGATTATCGCTGAGGATAATAAATTCCTTTTCATTTAATTGCCTTTTGTATATTTACGTTGGGGTTCTCTTATTTTAACGAACACATTGATATTGATAATGTTAATTATAATGAAATATGGATTAAGTATTTTTTGTGTTGAATAAATCACATACCTGTTTTTTTCATTAAAATGAATCTGTAAATTTCAATGATATTATTTATAGGTTATTCTAATGAAAAAGGATTTGTTATCATCAATTATTATCGCTATGCTGATGACTGCCGGGCTAAGCGCTTGTGATGAAAAAAAAGCGGACGAACAACCTGTAGCTCAGAGTGCGGATTCATCAGCGAGCAATACACAATCCACATCAGCGGAAAGCGCAGATGCGAACGATGTTCTTAACCAAAAGCTGAATGTTTACATTGATTGTTATAATAATCTCCAGGCAGATATTTACCGTGCTGTAAATCGTTATGCTAACACATTCGACGATTTCCGCACCGGTCCTACCGGTAAGGAAGACGATCCCTCTCCGCTGGTTCCTGTGTATCCTGCATTTATCCAGGACTGCCGGAAAGACATC</td></tr>
<tr class=alt_row><td><a name="barcode2_parC_v7_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>parC_v7</td><td>11</td><td></td><td>TGGCACAATCACTAAACGCGTCGGGTTTTCGTGATCCGATTCATCGCGCAGGTCGTCCACCATCGGCAGTTTTTTATTGCGCATCTGCGCAGCAATCTGCTCCAGCACTTTTGCGCCAGATACCTGATGCGGCAACGCGGTGATCACCACGGCGCCGTCTTCTTTGGTCCATACCGCGCGCATACGCACGGAGCCACGCCCATTTTCGTAAATTTTACGAATTTCCGCACGAGGGATAATGATCTCCGCTTCGGTCGGGTAATCCGGCCCCTGGACGATATCCAGCAACTGATCCAGCGTCGTTTTCGGCTGCTCAATCAGCGTAATCGCCGCTTTCGCGACTTCGCGCAGGTTATGCGGCGGAATATCTGTTGCCATGCCCACCGCAATACCGGTGGTGCCGTTCAGCAGGATGTTCGGCAGACGCGCCGGTAACATTTTCGGTTCCTGCATGGTGCCGTCGAAGTTTGGCACCCAGTCCGCAGTCCCCTGACCGAGTTCGCTTAACAGCAGCTCGGCGTATTTGGACAGGCGGGATTCGGTATAACGCATCGCCGCGAATGACTTCGGATCATCCGGCGCGCCCCAATTCCCCTGGCCATCGACCAGCGGGTAACGGTAAGAGAACGGCTGCGCCATCAGCACCATGGCTTCATAGCAGGCGCTGTCGCCATGCGGGTGATACTTACCCAGTACGTCG</td></tr>
<tr><td><a name="barcode2_tviD_v2_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>tviD_v2</td><td>298</td><td>Pos:430 (95%) , Change:C>T, Implication: known, but uninformative allele</td><td>GCAAGCTGCTTAGTGATCGATAAGGTAATCTCACGTATTTGGTCCTGTTTCTCTTGTCGCCCCTCATCAAGATGATGATTAGTATTATTGATCACTCCCTTAAAAATAGCCGCAACGATCGCGTCACCAAACATATCGGTATAATGAGTAAGATCGTCGCCATTTTTTTCCAGGGCTCGTTTCTGCCCCCACTTTTCCATCAAATTTGTCGGGTTAACGCAGGGAATATCCATCCTGGCGGCGATGGTGTCGACATTTTTAATCAACCGACTACGGGATAGAATGACGGTTCCAGCACGAGTCACGGCATCAACATGTGTCATAAACATAACGCGATCGCGACCAAGAAGCTGAACGATTTCCATCATATCCTGTTCGATAGCATGCTCATCCATCTGCTCGACATTGATATTGCTTAATAAGGCACGATTGTCTTCCGACATTCCGGCAAAACAGGGATCGTCTTTGAGATAGGCCAGCAGTTTGTGTCGATTTCCTTGCGTGGCCAACGACCAGTACATGCACGCCCGCTCTGTTTGGCTAAAAAATTCATAGAAATGACGAGTTAAATAATTTATTTGTAAGCAGAACCCATAGGCCATAATCTTCTTCTGACTTGAAATCTCAATGATATAGAAATCAGAACGACTATGTACGTCGGTGTTTGAATAGTTTACATTAGGTCTAAAAATAAATGGTCGGACCTTTTCAGGAATGTCTATCAAGCCCAAAATAAACCGAATCTGCTGTAATGCCTCGCTGCTCGTATGCGTAAAACCATATATTCTCTTAAAGTTTGCCTGAAAATTAAAATAAGGC</td></tr>
<tr class=alt_row><td><a name="barcode2_acrB_v2_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>acrB_v2</td><td>510</td><td></td><td>CACAGGAAGACGACGATTAGCGAGATAGCATACAGGGCAGGGGCCTGGTTGCCGGACAACCGCTCCTGGTAGGACATCCCGGTCCAGTCATACCCAATGCCTGACGGCAGCTTGCTGGCCAGTTCTTCCATCATCGCCATCGCTTCACCAGTACTCTTGCCTGGCGCCGCCTGACCCAGAATTTCCATCGAAGGCAGACCGTTATAGCGTTCCAGACGCGGCGAACCATATTCCCAGCGGGAAGAGGAGAATGCGGAGAACGGCACCATCTGACCATCGCTACCACGAACGTACCAGTCGTTAATATCATCCGGCAACATGCGGTATTTCGCTTCGGACATCACGTAAACTTTCTTCACACGACCGCGATCGATAAAGTCGTTTACATAGCTGCCGCCCCATGCTGCGCCCAGCGTGGTATTAATGTCGCTAATAGACACGCCCAGCGCCTGAGCTTTTTCCTGGTCGATATCGATTTTAAACTGCGGCGTATCTTCCAGACCGTTAGGTCGAACGCCGACCAACAGATCAGGATATTTCGCCACCTCGCCGAACAACTGATTACGTGCCTGGGTGAGTTTTTCGTGACCAAGTCCCGCCTGGTCAATCAACTCGAAGTCAAACCCGGTTGCGGTGCCCAGTTCAACAATCGCCGGTAGGTTAAAGGCGAAGACCATCGCATCTTTAATTTGTGAAAACGCTGCGGTTGCCCGCTGGGTAATCGCTTCAACCTTGTTTTTTTCGCCTGGACGATCGGCCCAGTCTTTCAACGACACAAATGCAATACCGGTATTCTGACCGCGCCCTGCAAAACCGAAGCCGTTGACGGCGAATACCGATTCAACGTTGGCTTTCTCTTTGTTCAGATAGTAATCCGTGACCTCATCCAGCACTT</td></tr>
<tr><td><a name="barcode2_gyrA_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>gyrA</td><td>93</td><td>Pos:320 (97%) , Change:C>T, Implication: Unknown<br/>Pos:1684 (99%) , Change:G>T, Implication: gyrA_S83T</td><td>GACGCCTTCTTCGTACTCACGAACCGGCAGAATCGCGGTGATACGTTCGTTGGCTTCCAGCGGCAGCAGGTTAACGATCGGACGACCGCGCGCGCCACGGCTGGCTTCCGGCAACTGATAGACCTTCATCCAGTACAGACGGCCCCGGCTGGAGAAGCAAAGGATGGTGTCATGGGTGTTGGCCACCAGCAGGCGGTCGATAAAGTCTTCTTCTTTAATACGTGCTGCTGATTTGCCTTTACCGCCACGACGCTGTGCTTCGTAATCGGTTAACGGCTGATATTTCACATAGCCCTGATGAGACAAGGTTACGACAACATTTTCCTGGCTAATCAGATCTTCGATATTAATATCGGCGCTGTTGGCGGTGATTTCGGTACGACGCTCATCGCCGAACTGATCGCGAATTAACTCCATCTCTTCGCGGATCACTTCCATCAGGCGATCGGCGCTGCCCAGAATGTGCAGCAGTTCAGCGATCTGCTCCAGCAACTCTTTGTATTCGTCGAGCAGTTTTTCATGCTCCAGGCCGGTCAGTTTCTGCAAACGCAGATCCAGAATCGCCTGCGCCTGCTGTTCAGTCAGGTAGTATTGACCGTCACGCACGCCAAATTCTGGCTCCAGCCATTCCGGACGCGCGGCGTCGTCACCGGCGCGCTCCAGCATCGCAGCAACGTTGCCCAGATCCCACGGACGCGAAATCAGCGCCGCTTTCGCTTCCGCCGGCGTCGGCGCGCGGCGAATCAGTTCGATAATCGGGTCGATGTTGGCCAGCGCAATCGCCAGAGCTTCAAGGATATGCGCACGATCACGGGCTTTACGCAGTTCAAAAATAGTCCGACGCGTCACCACTTCACGGCGGTGGCGCACGAACGCTGAAATAATATCTTTCAGGTTCATGATCTTCGGCTGGCCGTGATGCAGCGCCACCATGTTAATACCGAAGGAAACCTGTAGCTGGGTCTGGGAGTAGAGATTATTAAGCACCACCTCGCCCACCGCATCGCGTTTCACTTCAATCACGATGCGCATCCCGTCTTTGTCGGATTCGTCACGCAGCGCGCTGATGCCTTCCACGCGTTTATCTTTCACCAGCTCGGCGATTTTCTCGATCAGGCGCGCTTTGTTCACCTGATAGGGAATTTCATGGACGATGATGGTTTCACGGCCCGTTTTCGCGTCAGCTTCAACTTCCGCGCGGGCGCGAATGTACACTTTGCCACGACCGGTACGGTAGGCTTCTTCGATACCACGACGACCGTTGATGATCGCGGCGGTCGGGAAGTCCGGACCCGGAATATGTTCCATCAGCCCTTCAATGCTGATGTCTTCGTTGTCGATATACGCCAGGCAGCCGTTAATCACTTCCGTCAGGTTGTGCGGCGGGATATTCGTCGCCATACCTACTGCGATACCGGAAGAACCGTTCACCAGCAGATTCGGAATTTTGGTCGGCATGACGTCCGGAATTTTTTCCGTACCGTCATAGTTATCCACGAAATCCACCGTCTCTTTTTCGAGATCGGCCATCAGTCCGTGGGCGATTTTCGCCAGACGGATCTCCGTATAACGCATTGCCGCCGCGGAGTCGCCGTCAATAGAACCGAAGTTACCCTGACCATCCACCAGCATGTAACGCAGCGAGAATGGCTGCGCCATACGAACGATGGTGTCATACACTGCGTAATCGCCGTGGGGATGGTATTTACCGATTACGTCACCAACGACACGGGCAGATTTTTTATAGGCTTTGTTCCAGTCATTGCCCAATACGTTCATGGCGTAAAGTACGCGACGGTGTACCGGCTTCAGGCCATCTCGGACATCCGGCAGCGCACGGCCAACAATGACAGACATCGCATAATCCAGATAGGAGCTCTTCAGCTCTTCCTCAATGTTGACCGGTGTAATTTCTCTCGCAAGGTCGCTCATCTAACCGCTATCCCTCTACTGTATCCCGGATTCAAAGGTCGCAAATTATAACACATTCGCCCACGTACTGGTAAACGGAATGCAGCCAGACAATGCTTTATTCGGGTGCCACATGCTCTCTTCATTTATAAGCGTGGTCTCGGCGTTTATCGGCGATCAGCTTCA</td></tr>
<tr class=alt_row><td><a name="barcode2_plasmid_ST6_C19241A_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>plasmid_ST6_C19241A</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode2_chromosomal_MDR_cyaA_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>chromosomal_MDR_cyaA</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a name="barcode2_chromosomal_MDR_yidA_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>chromosomal_MDR_yidA</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode2_plasmid_MDR_AL513383.1_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>plasmid_MDR_AL513383.1</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a name="barcode2_plasmid_MDR_CP029645.1_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>plasmid_MDR_CP029645.1</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode2_plasmid_MDR_CP029879.1_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>plasmid_MDR_CP029879.1</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a name="barcode2_plasmid_MDR_CP029924.1_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>plasmid_MDR_CP029924.1</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode2_plasmid_MDR_CP029957.1_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>plasmid_MDR_CP029957.1</td><td>0</td><td></td><td></td></tr>
<tr class=alt_row><td><a name="barcode2_plasmid_MDR_LT904879.1_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>plasmid_MDR_LT904879.1</td><td>0</td><td></td><td></td></tr>
<tr><td><a name="barcode2_plasmid_MDR_LT904892.1_consensus" href="#barcode2">barcode2<br/>S. Typhi H58 + wastewater</a></td><td>plasmid_MDR_LT904892.1</td><td>0</td><td></td><td></td></tr>
	</tbody>
</table>
</div>
<div class=header_line><a name="ModelSignatures">Model Signatures</a><div><a href="#Summary">Back to Summary</a></div></div>
<br/>
<table>
	<tbody>
<tr class=table_header><td>Amplicon</td><td>Is Core</td><td>Signature</td><td>Date Trained</td></tr>
		<tr class=alt_row><td>1_3_4.3.1</td><td>True</td><td>c6a124cc-2d1a-439d-b218-82f0169d7dce</td><td>17/Jun/24 at 14:11</td></tr>
		<tr><td>2.2.2_2</td><td>True</td><td>d42fe4cc-a399-4939-afc7-a91b1a1d78ba</td><td>17/Jun/24 at 14:14</td></tr>
		<tr class=alt_row><td>2.3.1_v12</td><td>True</td><td>05c89613-314d-4021-9f7f-450db4e3d6bf</td><td>17/Jun/24 at 14:10</td></tr>
		<tr><td>2.3.2</td><td>True</td><td>3760123e-71b1-45d4-a81e-f514999694f2</td><td>17/Jun/24 at 14:15</td></tr>
		<tr class=alt_row><td>2.5_4.3.1.2.1_3.3</td><td>True</td><td>39f7a587-e9d6-4e67-8d3e-5bac1ac29d81</td><td>17/Jun/24 at 14:16</td></tr>
		<tr><td>3.1.1_v5</td><td>True</td><td>ff776cf6-a3bb-4622-bf05-e48620629afe</td><td>17/Jun/24 at 14:16</td></tr>
		<tr class=alt_row><td>3.3.1</td><td>True</td><td>696621aa-ab6c-4056-8825-0740f6006db1</td><td>17/Jun/24 at 14:16</td></tr>
		<tr><td>4.3.1.1</td><td>True</td><td>52575983-1e91-4f23-b792-095e94a5d30d</td><td>17/Jun/24 at 14:17</td></tr>
		<tr class=alt_row><td>4.3.1.1.P1_v2</td><td>True</td><td>599113c8-2f29-4cfc-98cd-95b14bdc656c</td><td>17/Jun/24 at 14:17</td></tr>
		<tr><td>4.3.1.2.1.1_4</td><td>True</td><td>983bccd2-b31a-45c8-ae85-6380b891601b</td><td>17/Jun/24 at 14:18</td></tr>
		<tr class=alt_row><td>4.3.1.2_v3</td><td>True</td><td>52e276eb-b247-45e6-8d99-12328a8c031d</td><td>17/Jun/24 at 14:18</td></tr>
		<tr><td>acrB_v2</td><td>True</td><td>ce95b108-2039-48a5-8026-49de40cd4734</td><td>17/Jun/24 at 14:19</td></tr>
		<tr class=alt_row><td>chromosomal_MDR_cyaA</td><td>False</td><td>d48a9018-4a6d-4666-90a8-0afcf5d8c10c</td><td>17/Jun/24 at 15:08</td></tr>
		<tr><td>chromosomal_MDR_yidA</td><td>False</td><td>20dbc1d7-4a1f-47db-a94a-f0199c93ca2c</td><td>17/Jun/24 at 15:23</td></tr>
		<tr class=alt_row><td>gyrA</td><td>True</td><td>63c9a7d3-1ed2-48eb-9e75-1e61f850e827</td><td>17/Jun/24 at 14:20</td></tr>
		<tr><td>parC_v7</td><td>True</td><td>0e26df39-abe8-45e7-86b1-215c43eb232c</td><td>17/Jun/24 at 15:05</td></tr>
		<tr class=alt_row><td>plasmid_MDR_AL513383.1</td><td>False</td><td>b58fd6b7-b7d0-483c-af7a-2560d0cd76f5</td><td>17/Jun/24 at 15:58</td></tr>
		<tr><td>plasmid_MDR_CP029645.1</td><td>False</td><td>cbfed9cf-b65b-4d52-96d7-df4bc4d8755a</td><td>17/Jun/24 at 16:05</td></tr>
		<tr class=alt_row><td>plasmid_MDR_CP029879.1</td><td>False</td><td>78c9f944-6bad-4609-bc5e-2246e99a9e35</td><td>17/Jun/24 at 16:26</td></tr>
		<tr><td>plasmid_MDR_CP029924.1</td><td>False</td><td>de9bd183-f458-4df1-bb51-c910db5130d0</td><td>17/Jun/24 at 17:29</td></tr>
		<tr class=alt_row><td>plasmid_MDR_CP029957.1</td><td>False</td><td>2f098d3f-8505-48cd-b12d-0731a746d15e</td><td>17/Jun/24 at 17:38</td></tr>
		<tr><td>plasmid_MDR_LT904879.1</td><td>False</td><td>126fbedc-967d-404d-b131-0e1cced3a679</td><td>17/Jun/24 at 16:09</td></tr>
		<tr class=alt_row><td>plasmid_MDR_LT904892.1</td><td>False</td><td>c14ee0aa-de18-4fc7-8196-641b6613ae06</td><td>17/Jun/24 at 15:31</td></tr>
		<tr><td>plasmid_ST6_C19241A</td><td>True</td><td>1bddb687-fd77-40a7-8183-4a3a39d735be</td><td>18/Dec/24 at 12:26</td></tr>
		<tr class=alt_row><td>tviD_v2</td><td>True</td><td>3149a8ad-4c47-41de-91ed-1002f41f759c</td><td>17/Dec/24 at 18:13</td></tr>
<table>
	<tbody>
</div>
</body>
</html>
